src/Embedders/Word2Vec.php

```diff
@@ -0,0 +1,920 @@
+        }
+
+        if ($minCount < 1) {
+            throw new InvalidArgumentException("Minimum word count must be greater than 0, $minCount given.");
+        }
+
+        $this->layer = $layer;
+        $this->window = $window;
+        $this->dimensions = $dimensions;
+        $this->sampleRate = $sampleRate;
+        $this->alpha = $alpha;
+        $this->epochs = $epochs;
+        $this->minCount = $minCount;
+        $this->negLabels = Vector::quick([1, 0]);
+        $this->sigmoid = new Sigmoid();
+    }
+
+    /**
+     * Return the data types that this estimator is compatible with.
+     *
+     * @return \Rubix\ML\DataType[]
+     */
+    public function compatibility() : array
+    {
+        return [
+            DataType::categorical(),
+        ];
+    }
+
+    /**
+     * Return the settings of the hyper-parameters in an associative array.
+     *
+     * @return mixed[]
+     */
+    public function params() : array
+    {
+        return [
+            'layer' => $this->layer,
+            'window' => $this->window,
+            'dimensions' => $this->dimensions,
+            'sample_rate' => $this->sampleRate,
+            'alpha' => $this->alpha,
+            'epochs' => $this->epochs,
+            'min_count' => $this->minCount,
+        ];
+    }
+
+    /**
+     * Has the learner been trained?
+     *
+     * @return bool
+     */
+    public function fitted() : bool
+    {
+        return !empty($this->vectors);
+    }
+
+    /**
+     * Iterating through a range of the specific epoch count and updating all respective word vectors.
+     *
+     * @param \Rubix\ML\Datasets\Dataset $dataset
+     * @throws \InvalidArgumentException
+     */
+    public function fit(Dataset $dataset) : void
+    {
+        DatasetIsNotEmpty::check($dataset);
+
+        $sentences = $dataset->column(0);
+
+        $this->preprocess($sentences);
+        $this->prepareWeights();
+
+        $startAlpha = $this->alpha;
+
+        for ($i = 0; $i < $this->epochs; ++$i) {
+            $this->alpha = $startAlpha - (($startAlpha - self::MIN_ALPHA) * ($i) / $this->epochs);
+
+            $this->trainEpochSg();
+        }
+
+        $this->generateL2Norm();
+    }
+
+    /**
+     * Embed a high dimensional dataset into a lower dimensional one.
+     *
+     * @param array[] $samples
+     * @throws \RuntimeException
+     */
+    public function transform(array &$samples) : void
+    {
+        if (!$this->vectorsNorm) {
+            throw new RuntimeException('Transformer has not been fitted.');
+        }
+
+        foreach ($samples as &$sample) {
+            foreach ($sample as $i2 => $sentence) {
+                $preppedSentence = $this->prepSentence($sentence);
+
+                $embeddings = [];
+                foreach ($preppedSentence as $word) {
+                    $embeddings[] = $this->embedWord($word);
+                }
+
+                $sample[$i2] = Matrix::stack($embeddings)->transpose()->mean();
```

Comment:
This is replacing the feature column at `$i2` (probably should be called `$j` since it's the second inner loop) with a column vector. We do not store vectors (or other arrays for that matter) in the feature columns of a sample matrix. Instead, concatenate the word vectors produced by W2V with the non-categorical features of the sample making sure to unset the old text blob feature column.